File: CSharp/GenericStack/StackNode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GenericStack
{
    class StackNode<T>
    {
        public T data { get; set; }

        public StackNode<T> next { get; set; }

        public StackNode(T dataValue, StackNode<T> nextNode)
        {
            data = dataValue;
            next = nextNode;
        }

        public StackNode(T dataValue) : this(dataValue,null) { }
    }
}

File: Java/src/genericStack/StackNode.java
package genericStack;

public class StackNode<T> {
	
	public T data;

    public StackNode<T> next;

    public StackNode(T dataValue, StackNode<T> nextNode)
    {
        data = dataValue;
        next = nextNode;
    }

    public StackNode(T dataValue){
    	data = dataValue;
        next = null;
    }

}
